apps/ecommerce/lambda/src/controllers/apiController.ts

```diff
@@ -1,25 +1,33 @@
 import { Request, Response } from 'express';
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import { HydratedResourceData, ResourceLink } from '@/src/types';
+import { config } from '../config';
 
 const ApiController = {
   ping: (req: Request, res: Response) => {
     return res.send({ status: 'ok', message: 'pong' });
   },
-  product: (req: Request, res: Response) => {
-    console.log(req.params);
-    return res.send({
-      sys: {
-        type: 'ResourceLink',
-        linkType: 'IntegrationResource',
-        urn: 'crn:shopify:::product:products/8191006998814',
-      },
-      name: 'Metallica T Shirt',
-      description: "An awesome men's T-shirt with metallica on it",
-      status: 'Out of stock',
-      image: 'https://cdn.shopify.com/images/foobar.jpg',
-      extras: {
-        sku: 'abc123',
-      },
-    });
+  resource: async (
+    req: Request,
+    res: Response<HydratedResourceData | { error: any; message: string }>
+  ) => {
+    const body: ResourceLink = req.body;
+    const BASE_URL = config.baseUrl;
+    try {
+      const url = `${BASE_URL}/${body.sys.provider.toLowerCase()}/resource/${body.sys.linkType}/${
+        req.params.id
+      }`;
+
+      const response: AxiosResponse<HydratedResourceData> = await axios.post(url, body);
```

Comment:
@pkeavenycontentful i think we can more of less hard code the URL to the proxy to avoid the VULN introduced here that CodeQL is flagging. Something like a mapping of providers to proxy URL 


```
const PROVIDERS = {
  shopify: `${config.baseUrl}/shopify/resource`
}
```

and then `axios.post(PROVIDERS[req.body.sys.provider.toLowerCase(), req.body)` perhaps? we can toy around with this if we want to, just spit balling here

Reply:
- [ ] refactor to avoid SSRF vuln

Reply:
eventually we can move this to a client wrapper which handles the params etc. so i'd support whatever easy fix allows us to avoid the vulneratiblity.

Reply:
yep absolutely